framework/src/vectorpostprocessors/PositionsFunctorValueSampler.C

```diff
@@ -63,11 +63,14 @@ PositionsFunctorValueSampler::execute()
   const auto state = determineState();
 
   // Pull new points
+  const auto old_size = _points.size();
   _points = _positions.getPositions(_fe_problem.getCurrentExecuteOnFlag() == EXEC_INITIAL);
+  if (_points.size() != old_size)
+    mooseError("The points array has grown since initialize() was called");
```

Comment:
This is definitely an improvement. Of course you could have the same number of points and have them be at different locations. I would maybe propose adding a `DEBUG` mode check that compares the vectors for equality

Reply:
I dont think that s a problem. It just means something updates the positions, but the containers to receive the information are still the right size